realm-jni/build.gradle

```diff
@@ -87,16 +87,39 @@ if (project.forceDownloadCore instanceof String) {
     ext.forceDownloadCore = Boolean.parseBoolean(project.forceDownloadCore)
 }
 
-task checkProperties(group: 'check', description: 'Check the user provided gradle.properties') << {
-    if (!project.ext['ndk.dir']) {
-        throw new GradleException('The ndkDir property in the gradle.properties file is not set.')
-    } else if (project.ext['ndk.dir'].endsWith('/')) {
-        throw new GradleException('The path provided in the ndkProperty in the gradle.properties ends with /')
-    } else if (!file(project.ext['ndk.dir']).directory) {
-        throw new GradleException('The path provided in the ndkDir property in the gradle.properties is not a folder.')
-    } else if (!file(new File(project.ext['ndk.dir'], 'RELEASE.TXT')).file) {
-        throw new GradleException('The path provided in the ndkDir property in the gradle.properties does not seem to be an Android NDK.')
+def getNdkDir() {
+    def ndkDir = null
+    if (project.ext.properties.containsKey('ndk.dir')) {
+        ndkDir = project.ext['ndk.dir']
+    } else if (System.getenv('NDK_HOME')) {
+        ndkDir = System.getenv('NDK_HOME')
+    }
+    if (ndkDir && ndkDir[-1] == "/") {
+        ndkDir = ndkDir[0..-2]
+    }
+    return ndkDir
+}
+
+def verifyNdk() {
+    def ndkDir = getNdkDir()
+    if (!ndkDir) {
+        throw new GradleException('The ndk.dir property in the local.properties file and NDK_HOME are not set. One of them should be set.')
+    }
+    if (!file(ndkDir).directory) {
+        throw new GradleException('The path provided in the ndk.dir property in the local.properties or NDK_HOME is not a folder.')
     }
+    def releaseFile = new File("${ndkDir}/RELEASE.TXT")
+    if (!releaseFile.file) {
+        throw new GradleException('The path provided in the ndk.dir property in the local.properties or NDK_HOME does not seem to be an Android NDK.')
+    }
+    def ndkVersion = releaseFile.getText('UTF-8').split()[0]
+    if (ndkVersion != 'r10d') {
+        throw new GradleException("Your NDK version: ${ndkVersion}. Realm JNI should be compiled with the version r10d of NDK.")
+    }
+}
+
+task checkProperties(group: 'check', description: 'Check the user provided gradle.properties') << {
+    verifyNdk()
```

Comment:
It's in generateNdkToolchain${toolchain.name.capitalize()}" tasks, not in checkProperties task. I think it's not in the same scope. If we hope avoid calling `getNdkDir()` there, `ndkFolder` should be a global variable.